### PR TITLE
Upgrade OWASM Studio to support OWASM v0.1.4

### PR DESCRIPTION
**Pull request by smiled0g**
Change summary
- [x] Rewrite server script, leaving it with only essential build functionalities
- [x] Migrate OWASM local execution to use BandSV
- [x] Support code deployment to D3N testnet
- [x] Support code upload to S3

### CODE REVIEW

**Review by sorawit — DISMISSED** [review]

File: studio/server/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.DS_Store
```
Comment (sorawit) [inline studio/server/.gitignore]: This is not necessary.

File: studio/server/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+node_modules
```
Comment (sorawit) [inline studio/server/.gitignore]: Let's be specific: `/node_modules/`

File: studio/server/package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "owasm-studio-server",
+  "version": "1.0.0",
+  "description": "Compiler server for OWASM Studio",
+  "main": "index.js",
+  "author": "Band Protocol <dev@bandprotocol.com>",
+  "license": "MIT",
+  "private": true,
+  "scripts": {
+    "start": "nodemon ."
```
Comment (sorawit) [inline studio/server/package.json:10]: ???

File: studio/ui/templates/owasm/src/logic.rs [hunk 4]
```diff
@@ -29,24 +90,52 @@ mod tests {
...
+        let data = Data::build_from_local_env(&Parameter {
+            crypto_symbol: coins::Coins::ETH,
+            stock_symbol: String::from("FB"),
+            alphavantage_api_key: String::from("WVKPOO76169EX950"),
+        })
+        .unwrap();
+        println!("Current ETH and FB price (times 100) is {:?}", execute(vec![data]));
     }
-}
+}
```
Comment (evilpeach) [inline studio/ui/templates/owasm/src/logic.rs]: add line

File: studio/ui/templates/owasm/src/lib.rs [hunk 5]
```diff
@@ -22,24 +40,144 @@ pub fn __allocate(size: usize) -> *mut u8 {
...
+    //     assert_eq!(result, r#"[["coin_gecko","f32"],["crypto_compare","f32"]]"#);
+    // }
+
+    // #[test]
+    // fn test_params_info() {
+    //     let ptr_output = __params_info();
+    //     let result = std::str::from_utf8(__read_data(ptr_output)).unwrap();
+    //     assert_eq!(result, r#"[["symbol_cg","String"],["symbol_cc","String"]]"#);
+    // }
+}
```
Comment (evilpeach) [inline studio/ui/templates/owasm/src/lib.rs]: add line

File: studio/server/src/system.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
+const fs = require('fs')
+const cp = require('child_process')
```
Comment (sorawit) [inline studio/server/src/system.js:2]: This may be helpful https://www.npmjs.com/package/child-process-promise

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**